File: Aprendendo-C/Exercício/diagonal_negativos.c
#include <stdio.h>
#include <string.h>
#include "gabsLib.h"

int main(){
    printf("Qual a oredem da matriz? ");
    int n; scanf("%i", &n);

    int mat[n][n], vetDia[n], contNeg;

    for (int i = 0; i < n; i++){
        for (int j = 0; j < n; j++){

            printf("Elemento [%i,%i]: ", i, j );
            scanf("%i",&mat[i][j]);
        }
    }

    for (int i = 0; i < n; i++){
        for (int j = 0; j < n; j++){
            if (i == j){
                vetDia[i] = mat[i][j];
            }

            if (mat[i][j]< 0)
            {
                contNeg++;
            } 
        }
    }
    printf("Diagonal principal:"); 
    for (int j = 0; j < n; j++){

    printf("%i \n", vetDia[j]);    
    } 

    printf("Quantidade de negativos: %i", contNeg); 
    
    jumpln();
    return 0;
}

File: Aprendendo-Python/Exercícos/diagonal_negativos.py

n = int(input("Ordem da matriz: "))

matriz: [[]] = [[0 for x in range(n)] for x in range(n)]

neg = 0

for i in range(0,n):
    for j in range(0,n):
        matriz[i][j] = int(input(f"Elemento[{i},{j}]: "))

print("Diagonal Principal: ")
for i in range(0,n):
    for j in range(0,n):
        if i == j:
            print(f"{matriz[i][j]}", end=" ")


print()
for i in range(0,n):
    for j in range(0,n):
        if matriz[i][j] < 0:
            neg = neg+1

print(f"Quantidade de negativo: {neg}")

